mitmproxy/addons/tlsconfig.py

```diff
@@ -161,45 +158,40 @@ def create_proxy_server_ssl_conn(self, tls_start: tls.TlsStartData) -> None:
             else:
                 server.alpn_offers = tls.HTTP1_ALPNS
 
-        # We pass through the list of ciphers send by the client, because some HTTP/2 servers
-        # will select a non-HTTP/2 compatible cipher from our default list and then hang up
-        # because it's incompatible with h2.
-        if not server.cipher_list:
-            if ctx.options.ciphers_server:
-                server.cipher_list = ctx.options.ciphers_server.split(":")
-            elif client.cipher_list:
-                # We used to filter for known ciphers here, but that doesn't seem to make sense.
-                # According to OpenSSL docs, the control string str should be universally
-                # usable and not depend on details of the library configuration (ciphers compiled in).
-                server.cipher_list = list(client.cipher_list)
-
-        args = net_tls.client_arguments_from_options(ctx.options)
-
-        client_certs = args.pop("client_certs")
-        client_cert: Optional[str] = None
-        if client_certs:
-            client_certs = os.path.expanduser(client_certs)
-            if os.path.isfile(client_certs):
+        if not server.cipher_list and ctx.options.ciphers_server:
+            server.cipher_list = ctx.options.ciphers_server.split(":")
+        # don't assign to client.cipher_list, doesn't need to be stored.
+        cipher_list = server.cipher_list or DEFAULT_CIPHERS
+
+        client_cert: Optional[Path] = None
+        if ctx.options.client_certs:
+            client_certs = Path(ctx.options.client_certs).expanduser()
+            if client_certs.is_file():
                 client_cert = client_certs
             else:
                 server_name: str = (server.sni or server.address[0].encode("idna")).decode()
-                path = os.path.join(client_certs, f"{server_name}.pem")
-                if os.path.exists(path):
-                    client_cert = path
-
-        args["cipher_list"] = ':'.join(server.cipher_list) if server.cipher_list else None
-        ssl_ctx = net_tls.create_client_context(
-            cert=client_cert,
-            sni=server.sni.decode("idna") if server.sni else None,  # TODO: Should pass-through here.
+                p = (client_certs / f"{server_name}.pem")
```

Comment:
I see this code just moved down a bit, but just to raise the question: is `server_name` always a valid file name? Win/Linux/Mac filesystem restrictions and allowed characters and length?

Reply:
I'm sure you can find an edge case, but this should be fine. We'd need a user who needs client certificates for a _really_ weird domain.